src/Components/DataHistoryExplorer/Internal/TimeSeriesViewer/Internal/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {
+    useCallback,
+    useContext,
+    useEffect,
+    useMemo,
+    useRef,
+    useState
+} from 'react';
+import {
+    ITimeSeriesChartProps,
+    ITimeSeriesChartStyleProps,
+    ITimeSeriesChartStyles
+} from './TimeSeriesChart.types';
+import { getStyles } from './TimeSeriesChart.styles';
+import { classNamesFunction, useTheme, styled } from '@fluentui/react';
+import HighChartsWrapper from '../../../../../HighChartsWrapper/HighChartsWrapper';
+import ChartCommandBar from './Internal/ChartCommandBar/ChartCommandBar';
+import { IHighChartSeriesData } from '../../../../../HighChartsWrapper/HighChartsWrapper.types';
+import { transformADXTimeSeriesToHighChartsSeries } from '../../../../../../Models/SharedUtils/DataHistoryUtils';
+import { DataHistoryExplorerContext } from '../../../../DataHistoryExplorer';
+import { useTimeSeriesData } from '../../../../../../Models/Hooks/useTimeSeriesData';
+import { usePrevious } from '@fluentui/react-hooks';
+import { IDataHistoryChartOptions } from '../../../../../../Models/Types/Generated/3DScenesConfiguration-v1.0.0';
+import { QuickTimeSpanKey } from '../../../../../../Models/Constants/Enums';
+import { QuickTimeSpans } from '../../../../../../Models/Constants/Constants';
+
+const getClassNames = classNamesFunction<
+    ITimeSeriesChartStyleProps,
+    ITimeSeriesChartStyles
+>();
+
+const TimeSeriesChart: React.FC<ITimeSeriesChartProps> = (props) => {
+    const { timeSeriesTwinList, styles } = props;
+
+    // state
+    const [chartOptions, setChartOptions] = useState<IDataHistoryChartOptions>({
+        yAxisType: 'independent',
+        defaultQuickTimeSpanInMillis:
+            QuickTimeSpans[QuickTimeSpanKey.Last15Mins],
+        aggregationType: 'avg'
+    });
+
+    // hooks
+    const { adapter } = useContext(DataHistoryExplorerContext);
+    const xMinDateInMillisRef = useRef<number>(null);
+    const xMaxDateInMillisRef = useRef<number>(null);
+    const {
+        query,
+        deeplink,
+        data,
+        fetchTimeSeriesData,
+        isLoading
+    } = useTimeSeriesData({
+        adapter,
+        connection: adapter.getADXConnectionInformation(),
+        quickTimeSpanInMillis: chartOptions.defaultQuickTimeSpanInMillis,
+        twins: timeSeriesTwinList
+    });
+    const highChartSeriesData: Array<IHighChartSeriesData> = useMemo(
+        () =>
+            timeSeriesTwinList?.length
+                ? transformADXTimeSeriesToHighChartsSeries(
+                      data,
+                      timeSeriesTwinList
+                  )
+                : [],
+        [data, timeSeriesTwinList]
+    );
+
+    // callbacks
+    const updateXMinAndMax = useCallback(() => {
```

Comment:
this is not only the place which causes different snapshots, also the getMockTimeSeriesDataArrayInLocalTime method which returns random mock data points. if we set the nowInMillis to a constant, it will not be representing the real use case of the quick time dropdown interaction in the chart. Because it is supposed to show the "Last x" from time.now. If I set it to constant, in chromatic people would see charts like x axis ending at lets say Jan 1, 2023 even if the picker says "Last 15 minutes" which is supposed to be ending "now". But is it is okay to see weird behaviors in storybook, with `isChromatic` check I can make this change. Thanks for finding this!

What is you input also @darsney? The issue is since we need to use a changing x axis values due to quick time selection (Last x...) and data points, the snapshots are changing which will cause UI test differences in each PR. Mitch is suggesting using static data in stories. But what I am suggesting is in above paragraph ^. But I can definitely change this if you both think this is what it should be.

Reply:
okay I am going to use static values and implement this isChromatic logic in later PR